core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -497,35 +498,38 @@ sealed abstract class Cause[+E] extends Product with Serializable { self =>
 
   final def nonEmpty: Boolean = !isEmpty
 
-  /**
-   * Returns a `String` with the cause pretty-printed.
-   */
+  /** Returns a `String` with the cause pretty-printed. */
   final def prettyPrint: String = {
-    import Cause.Unified
+    val builder = new StringBuilder
+    prettyPrintWith(builder.append(_).append('\n'))(Unsafe.unsafe)
+    builder.result()
+  }
 
-    val builder = ChunkBuilder.make[String]()
-    var size    = 0
+  /** Pretty-prints this cause with the provided `println` function. */
+  final def prettyPrintWith(println: String => Unit)(implicit unsafe: Unsafe): Unit = {
```

Comment:
Could we change the argument name to something different? Calling `println` looks very wrong